tools/view_control_loop.py

```diff
@@ -43,33 +43,28 @@
     initialized = initBoards(client, board_ids)
 
     client.leaveBootloader(board_ids)
-    
     client.resetInputBuffer()
 
     initMotor(client, board_ids)
 
-    def updateCurrent(i): 
+    def updateCurrent(i):
         data = []
         for board_id in board_ids:
             try:
                 driveMotor(client, board_ids, actuations, mode)
                 # Read the iq calulated
-                read = struct.unpack('<f', client.readRegisters([board_id], [0x3003], [1])[0])
-                data.append(read)
+                data.append(struct.unpack('<f', client.readRegisters([board_id], [0x3003], [1])[0]))
                 # Read the iq command
-                read = struct.unpack('<f', client.readRegisters([board_id], [0x3020], [1])[0])
-                data.append(read)
-            except (ProtocolError, struct.error):
-                #print("Failed to communicate with board: ", board_id)
-                data.append([0.0])
-                data.append([0.0])
-        return time.time(), data
+                data.append(struct.unpack('<f', client.readRegisters([board_id], [0x3020], [1])[0]))
+            except (MalformedPacketError, ProtocolError):
+                print("Failed to communicate with board: ", board_id)
+        return time.time(), None if len(data) != 2 else data
```

Comment:
This function looks to be updating the currents for all boards, so we should expect 2 * (number of boards) in the data list?
```suggestion
        return time.time(), None if len(data) != 2 * len(board_ids) else data
```